handlers/wallet_service.py

```diff
@@ -0,0 +1,118 @@
+import json
+from typing import Optional, Union
+
+from aws_lambda_context import LambdaContext
+
+from common.errors import ApiError, RdsNotFoundError
+from usecases.wallet_service import WalletService
+from utils.wrappers.aws.api_gateway import ApiGateway, ApiGatewayEvent
+
+
+@ApiGateway()
+def handle_address_balance(
+    event: ApiGatewayEvent,
+    __: LambdaContext,
+    wallet_service: Union[WalletService, None] = None
```

Comment:
Why we dont use `Optional` here?

Reply:
`Optional[X]` and `Union[X, None]` are the same, so they can be used interchangeably but usually I prefer `Union[X, None]` for function args with default and properties.
This is not completely arbitrary, usually typing a parameter `Optional` gives the impression that if they are not present it will be `None` automatically, but you still have to add the `= None` after for the default value because `Optional` is just an alias.

That's why you will see more `Union` on class and function definitions and `Optional` on the function body.
But I'm not against just changing all to `Optional` since they are interchangeable. What do you think?

Reply:
Thank you for the explanation! I agree with maintaing `Union` here.